core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -155,6 +162,46 @@ sealed trait TLVFactory[+T <: TLV] extends Factory[T] {
   }
 }
 
+case class NormalizedString(private val str: String) {
+
+  val normStr: String = CryptoUtil.normalize(str)
+
+  override def equals(other: Any): Boolean = {
+    other match {
+      case otherStr: String =>
+        normStr == otherStr
+      case _ => other.equals(str)
+    }
+  }
+
+  override def toString: String = normStr
+}
+
+object NormalizedString {
```

Comment:
nit: Can we use `Factory[NormalizedString]`?

Reply:
Used `StringFactory` instead because `Factory` would then have the `apply(String)` function e the `fromHex` function instead of the contructor